src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/EditorConfig/Parsing/EditorConfigOption.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.EditorConfig.Parsing
+{
+    /// <summary>
+    /// Base option that all editorconfig option inherit from.
+    /// </summary>
+    internal record class EditorConfigOption(Section Section, TextSpan? Span);
```

Comment:
I originally did have it as `TextSpan` but in the course of implementing naming style parsing realized that there were cases where the options were implicit. For those cases I could just return `null` to imply "This option is not defined here" but in practice that meant that I would need to do extra work to lookup what the options default value _is_. I went with the approach of "This option has a value as far as we are concerned but it is implicit of the span is null".

Reply:
Not that for `Section` the span cannot be `null` because that doesn't make sense since all of them (even the implicit global section) have spans (even if that span is effectively zero length).